compiler-plugin/src/test/kotlin/arrow/meta/quotes/scope/TypeAliasTest.kt

```diff
@@ -1,27 +1,53 @@
 package arrow.meta.quotes.scope
 
+import arrow.meta.plugin.testing.Code
 import arrow.meta.plugin.testing.CompilerTest
 import arrow.meta.plugin.testing.CompilerTest.Companion.source
 import arrow.meta.plugin.testing.assertThis
+import arrow.meta.quotes.scope.plugins.TypeAliasPlugin
 import io.kotlintest.specs.AnnotationSpec
 
-// TODO Ast to Expr Conversion needed
-
 class TypeAliasTest : AnnotationSpec() {
 
-  private val typeAlias = """
+  @Test
+  fun `Validate type alias properties`() {
+    val typeAlias = """
                          | //metadebug
                          | 
-                         | typealias Predicate<T> = (T) -> Boolean
+                         | typealias IntegerPredicate = (Int) -> Boolean
                          | """.source
 
-  @Ignore
+    validate(typeAlias)
+  }
+
   @Test
-  fun `Validate type alias properties`() {
+  fun `Validate type alias with constraints properties`() {
+     val typeAlias = """
+                         | //metadebug
+                         | 
+                         | typealias Predicate<Int> = (Int) -> Boolean
+                         | """.source
+
+    validate(typeAlias)
+  }
+
+  @Ignore // issues with type erasure when AST parsing the compiled code?
+  @Test
+  fun `Validate type alias properties with generics`() {
+    val typeAlias = """
+                         | //metadebug
+                         | 
+                         | typealias Predicate<T> = (T) -> Boolean
+                         | """.source
+
```

Comment:
As expected, analysis for compilation erases generics at runtime.  @raulraja do you think there is a way to work around this?

Reply:
What is the difference in the output or exception stack-trace that makes the test fail?